### PR TITLE
Feat/authentication

### PR DESCRIPTION
**Pull request by sbracellari**

### CODE REVIEW

File: src/react/src/features/auth/Login.tsx [hunk 1]
```diff
@@ -21,49 +21,71 @@ export function Login() {
...
-            />
-            <Button
-                type="submit"
-                variant="contained"
-            >
-                Sign In
-            </Button>
-            {status === "failed" && message &&
-                <Alert sx={{ mt: 1 }} severity="error">{message}</Alert>}
+        <Box sx={{height: '100vh', display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
```
Comment (sbracellari) [inline src/react/src/features/auth/Login.tsx:28]: I just added a surrounding box and some styles and tabbed everything over. I didn't change the entire file lol

File: src/react/src/features/auth/Register.tsx [hunk 2]
```diff
@@ -26,58 +26,80 @@ export function Register() {
...
-                error={status === 'failed'}
-            />
-            <Button
-                type="submit"
-                variant="contained"
-            >
-                Sign Up
-            </Button>
-            {status === "failed" && message && <Alert severity="error">{message}</Alert>}
+        <Box sx={{height: '100vh', display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
```
Comment (sbracellari) [inline src/react/src/features/auth/Register.tsx:29]: See my comment on `Login.tsx`

File: src/react/src/features/router/Router.tsx [hunk 3]
```diff
@@ -23,23 +23,25 @@ export function Router() {
...
-            <Tabs value={tabValue} onChange={handleTabs} centered>
-                {tabComponent}
-            </Tabs>
             <Switch>
-                <Route exact path="/">
-                    <Redirect to="/collection" />
+                <Route exact path='/' component={Login} />
+                <Route exact path='/register' component={Register} />
+                <Route>
+                    <Tabs value={tabValue} onChange={handleTabs} centered>
```
Comment (sbracellari) [inline src/react/src/features/router/Router.tsx:33]: had to do some rearranging so that tabs wouldn't show up on login/register

Comment (dorisgjata) [reply src/react/src/features/router/Router.tsx:33]: So I think routes need to be tied to a path not wrap the elements

File: src/react/src/features/router/Router.tsx [hunk 4]
```diff
@@ -23,23 +23,25 @@ export function Router() {
...
+    //@todo redirect back to login if a user isn't logged in
     return (
         <BrowserRouter basename='/'>
-            <Tabs value={tabValue} onChange={handleTabs} centered>
-                {tabComponent}
-            </Tabs>
             <Switch>
-                <Route exact path="/">
-                    <Redirect to="/collection" />
+                <Route exact path='/' component={Login} />
```
Comment (dorisgjata) [inline src/react/src/features/router/Router.tsx:30]: path should be /login

File: src/react/src/services/api.ts [hunk 5]
```diff
@@ -62,104 +74,161 @@ export const api = createApi({
...
+                url: `consumer/favorites`,
+                method: 'GET',
+                mode: 'cors',
+                headers: {
+                    Authorization: 'Bearer ' + localStorage.getItem('token')
+                }
+            }),
         }),
         favoriteArtwork: builder.mutation({
             //@todo: how to define types for 2 params
```
Comment (dorisgjata) [inline src/react/src/services/api.ts:133]: can remove this todo since we only have 1 param now

File: src/react/src/services/auth.ts [hunk 6]
```diff
@@ -4,26 +4,40 @@ import { LoginUser, User } from "../app/types";
...
+    });
 }
 
 const login = (user: LoginUser) => {
-    return axios.post(API_URL + "signin", user).then((response) => {
-        console.log("login api", response);
-        if (response.data.accessToken) {
-            // @todo: should probably persist this in state too
-            localStorage.setItem("user", JSON.stringify(response.data));
+    return axios.get(API_URL + '/user/' + user.email + '/login').then((response) => {
```
Comment (dorisgjata) [inline src/react/src/services/auth.ts]:
```suggestion
    return axios.get(`${API_URL}/user/${user.email}/login`).then((response) => {
```

File: src/react/src/features/auth/Login.tsx (hunk 1 above)
Comment (dorisgjata) [reply src/react/src/features/auth/Login.tsx:28]: that's fine

File: src/react/src/features/auth/Register.tsx (hunk 2 above)
Comment (dorisgjata) [reply src/react/src/features/auth/Register.tsx:29]: We should probably unify the styles so we don't have pixels, rems and vh all together

File: src/react/src/features/router/Router.tsx (hunk 3 above)
Comment (sbracellari) [reply src/react/src/features/router/Router.tsx:33]: it seems to be working fine so far

File: src/react/src/features/router/Router.tsx (hunk 4 above)
Comment (sbracellari) [reply src/react/src/features/router/Router.tsx:30]: we can add a redirect to `/login`. but we need to show login at `/` too or else nothing will show up

Comment (dorisgjata) [reply src/react/src/features/router/Router.tsx:30]: That's what the redirect is for

**Comment by dorisgjata** [comment]
@Todos: 

- change `localStorage.get'user')` to `token` and update state
- update messages for login/register
- look up routes/paths and the redirect to /

**Pull request merged**